### PR TITLE
New documentation for TCP WAN communication [4391]

### PR DESCRIPTION
**Pull request by JuanCarlos-eProsima**
Create a new section to explain the TCP over WAN feature and how to use it.

### CODE REVIEW

File: docs/advanced.rst [hunk 1]
```diff
@@ -133,10 +138,70 @@ To configure the participant to connect to another node through TCP, you must ad
...
++---------------------------------------------------+
+
+The combination of the above configurations in both **TCP Server** and **TCP Client** allows a scenario similar to
+the represented by the following image.
+
+.. image:: TCP_WAN.png
+    :align: center
 
-Also, a TCP version of Hello-World example can be found in this `link <https://github.com/eProsima/Fast-RTPS/tree/master/examples/C%2B%2B/HelloWorldExampleTCP>`_.
+Finally, the concepts **TCP Server** and **TCP Client** only refer to how the TCP communication is established and
```
Comment (richiware) [inline docs/advanced.rst]: Move to the beginning of section, and improve the message of how works tcp transport.

File: docs/advanced.rst [hunk 2]
```diff
@@ -84,19 +84,24 @@ You can add custom transports using the attribute ``rtps.userTransports``.
...
+
 .. _comm-transports-tcp:
 
 TCP Transport
 =============
 
 To use TCP transports you need to define some more configurations:
 
-You must create a new TCP transport descriptor, for example TCPv4.
+You must create a new TCP transport descriptor, for example, TCPv4.
```
[the remark this answers is not in the document]
Comment (JuanCarlos-eProsima) [reply docs/advanced.rst]: removed

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/22/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/23/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/24/)

File: docs/advanced.rst [hunk 3]
```diff
@@ -84,19 +84,32 @@ You can add custom transports using the attribute ``rtps.userTransports``.
...
+All Transport configuration options can be found in the section :ref:`transportdescriptors`.
+
 .. _comm-transports-tcp:
 
 TCP Transport
 =============
 
+Unlike UDP, TCP transport is connection oriented and for that Fast-RTPS must establish a TCP connection
+before sending the RTPS messages.
+To do that there are two entities to know, **TCP Server** and **TCP Client**.
```
Comment (richiware) [inline docs/advanced.rst]:
```suggestion
Therefore TCP transport can have two behaviors, acts as a server (**TCP Server**) or as a client (**TCP Client**).
```

File: docs/advanced.rst [hunk 4]
```diff
@@ -84,19 +84,32 @@ You can add custom transports using the attribute ``rtps.userTransports``.
...
+The first one opens a TCP port listening for incoming connections and the second one tries to connect
+to the server. They don't have any relationship with RTPS concepts like **Publisher**, **Subscriber**,
```
Comment (richiware) [inline docs/advanced.rst]: I think it is not necesary this explication about the relatioship.

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/25/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/26/)

File: docs/advanced.rst [hunk 5]
```diff
@@ -84,19 +84,32 @@ You can add custom transports using the attribute ``rtps.userTransports``.
...
 .. _comm-transports-tcp:
 
 TCP Transport
 =============
 
+Unlike UDP, TCP transport is connection oriented and for that Fast-RTPS must establish a TCP connection
+before sending the RTPS messages.
+Therefore TCP transport can have two behaviors, acting as a server (**TCP Server**) or as a client (**TCP Client**).
+The server opens a TCP port listening for incoming connections and the client tries to connect
+to the server. The server and the client concepts are independent from the RTPS concepts: **Publisher**,
```
Comment (richiware) [inline docs/advanced.rst]: Several sentences in same line. Divide.

Comment (JuanCarlos-eProsima) [reply docs/advanced.rst]: Separated by each dot

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/28/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/29/)

**Pull request merged**